### ISSUE TITLE
A11y: Mobile nav button should be reachable via keyboard

### ISSUE THREAD

**Issue by bacitracin** [CodeForPhilly/clean-and-green-philly#978]
## Describe the task

When the screen is narrower and the mobile nav hamburger button pops up, it is not reachable or operable via keyboard navigation. Since it's not reachable by keyboard, screenreaders are affected as well.

This ticket is the first part of making the mobile nav more accessible. 

1. You should be able to TAB to the hamburger menu (probably need to add tabIndex=-1), 
2. You should be able to press ENTER or SPACE and the dropdown opens.

## Resources
Total expected behavior of the menu: You should be able to TAB to the hamburger menu, press ENTER or SPACE and the dropdown opens. Then you can use the UP or DOWN arrow keys to navigate the items (with focus state visible). Press ENTER or SPACE to choose one of the items. 

While the dropdown is open pressing ESC or TAB will close the dropdown and focus will go back to the hamburger button. 

See this example for most of the desired behavior (except for the behavior while the dropdown is open and pressing TAB.) https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/#mythical-page-content

**Comment by millmason** [comment]
🌤️ Hello! I believe I have a fix for this. Would you mind assigning it to me? @bacitracin

**Comment by bacitracin** [comment]
Assigned

**Comment by millmason** [comment]
Looking good on staging so far 🤞

**Comment by millmason** [comment]
Looking good on live! I think we're set to close this ticket.

**Comment by CodeWritingCow** [comment]
@millmason I'll close this ticket as complete!
